rfd/0203-database-healthchecks.md

```diff
@@ -0,0 +1,697 @@
+
+Recall that there may be multiple `db_server` heartbeats for the same database, but the web UI only shows a single resource tile for the database.
+If this is the case, then the tooltip will be shown if any of the heartbeats have an unhealthy `target_health.status`.
+The tooltip content may look something like this (or equivalent singular form verbiage if there's only one agent):
+
+    M out of N Teleport database services proxying access to this database cannot reach the database endpoint.
+    <link to healthcheck/network troubleshooting doc>
+
+    Affected Teleport database services:
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Error: <last health check error message>
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Error: <last health check error message>
+
+Databases with a mix of `healthy` and `""` (disabled) status will also show a tooltip that encourages the user to enable health checks or update to a supported version on all of the agents that proxy the db.
+The tooltip content may look something like this: 
+
+    M out of N Teleport database services proxying access to this database are not running network health checks for the database endpoint.
+    User connections will not be routed through affected Teleport database services as long as other database services report a healthy connection to the database.
+
+    Affected Teleport database services:
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Reason: The database service version v17.1.2 does not support health checks.
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Reason: The database service has disabled health checks for this database.
+
+#### User story: updating health check configuration
+
+Alice is a Teleport cluster admin who recently updated her cluster to a Teleport major version that supports health checks.
+
+After reading the changelog and supporting reference documentation for health checks, Alice checks the default settings using tctl:
+
+    $ tctl get health_check_config/default
+    kind: health_check_config
+    version: v1
+    metadata:
+      name: "default"
+      labels:
+        teleport.internal/resource-type: preset
+    spec:
+      enabled: true
+      match:
+        db_labels:
+          - name: '*'
+            values:
+            - '*'
+      interval: 10s
+      timeout: 5s
+      healthy_threshold: 2
+      unhealthy_threshold: 1
+    
+Out of an abundance of caution around this new feature, Alice wants to opt-out of health checks for prod databases.
+
+Alice uses `tctl edit health_check_config/default` to update the default settings and exclude prod databases:
+
+    kind: health_check_config
+    version: v1
+    metadata:
+      name: "default"
+      labels:
+        teleport.internal/resource-type: preset
+    spec:
+      enabled: true
+      match:
+        db_labels:
+          - name: '*'
+            values:
+            - '*'
+        db_labels_expression: "labels.env != `prod`"
```

Comment:
is this correct syntax for label expressions? should be something like `'labels["env"] != "prod"'`?

nit: i think we should encourage health check for prod envs. any other example we can use here?